samples/csharp_dotnetcore/58.qnamaker-multiturn-sample/BotServices.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.Bot.Builder.AI.QnA;
+using Microsoft.Extensions.Configuration;
+
+namespace Microsoft.BotBuilderSamples
+{
+    public class BotServices : IBotServices
```

Comment:
This is the same concept, I took from "NLP-With-Dispatch-Bot" sample. Point of keeping BotService was that, it can contain multiple QnA/LUIS services inside.

Reply:
Yes, but this approach, is really just a hang on from the botbuilder configuration work. In that model bot builder essentially invented its own configuration model and associated DI framework over and above what the platform provided. Over the last year we have moved away from this model in favor of leveraging the underlying platform configuration and DI support where it was relevant: after all bots are not different in this regard.

You are correct the nlp-with-dispatch follows the services model - that is not good and not something we should be propagating - it is something we should fix over time. For example, in that sample, couldn't both LUIS and QnA maker "services" be more directly modeled as regular dependencies added using regular mechanisms? The concept of "services" adds no value to the sample.

Removing the notion of "services" from our platform is just one less thing for the developer to have to understand. Of course, needless to say, an application developer is free to introduce such ideas themselves - its just that we think the most helpful thing is for our samples to be as narrow and direct about our platform as is possible.

Reply:
Done